security/pkg/nodeagent/caclient/credentials.go

```diff
@@ -66,8 +66,13 @@ func (t *TokenProvider) GetRequestMetadata(ctx context.Context, uri ...string) (
 		return nil, nil
 	}
 	if t.opts.TokenManager == nil {
+		if t.forCA {
+			return map[string]string{
+				"authorization": "Bearer " + token,
```

Comment:
I don't understand why in the client code, CA and XDS is differernt, but in the server code its the same. I don't see how this could work if you use Istiod as  CA

Reply:
If you use Istiod as CA, the header type is not controlled by XDS_TOKEN_TYPE, which is only applicable to XDS flow.